sources/packages/backend/libs/integrations/src/services/sfas/sfas-restriction-import.service.ts

```diff
@@ -26,65 +31,92 @@ export class SFASRestrictionImportService
   extends DataModelService<SFASRestriction>
   implements SFASDataImporter
 {
-  private readonly bulkInsertLegacyRestrictionsSQL: string;
   private readonly bulkInsertSFASMappedRestrictionsSQL: string;
   constructor(
     private readonly dataSource: DataSource,
     private readonly systemUsersService: SystemUsersService,
     private readonly notificationActionsService: NotificationActionsService,
     @InjectRepository(Restriction)
     private readonly restrictionRepo: Repository<Restriction>,
+    @InjectRepository(SFASRestrictionMap)
+    private readonly sfasRestrictionMapRepo: Repository<SFASRestrictionMap>,
   ) {
     super(dataSource.getRepository(SFASRestriction));
-    this.bulkInsertLegacyRestrictionsSQL = getSQLFileData(
-      "Bulk-insert-legacy-restrictions.sql",
-      SFAS_RESTRICTIONS_RAW_SQL_FOLDER,
-    );
     this.bulkInsertSFASMappedRestrictionsSQL = getSQLFileData(
       "Bulk-insert-sfas-mapped-restrictions.sql",
       SFAS_RESTRICTIONS_RAW_SQL_FOLDER,
     );
   }
 
+  /**
+   * Ensures that all the legacy only restrictions that are present in the
+   * sfas_restrictions_map table are also present in the restrictions table.
+   */
+  private async ensureLegacyOnlyRestrictionsExists(): Promise<void> {
+    // Create the query to identify the missing restriction codes to be inserted.
+    const existingRestrictionCodes = this.restrictionRepo
+      .createQueryBuilder("restriction")
+      .select("restriction.restrictionCode")
+      .getSql();
```

Comment:
What is suppose to be the outcome if a SIMS federal restriction code is added to mapping by mistake. I see that the system will not create new restriction on-demand. But will we insert a federal restriction as per the existing logic ?? Asking the question as configuration is dynamic now.

Reply:
Whatever is added to this table as "legacy only" will become a provincial restriction.
Can you clarify what is the concern here?
The table is added as a configuration option but should be managed with caution.